projects/libsodium/secretbox_easy_fuzzer.cc

```diff
@@ -0,0 +1,29 @@
+extern "C" {
+  #include <sodium.h>
+}
+
+class SodiumState {
+public:
+  unsigned char key[crypto_secretbox_KEYBYTES];
+  unsigned char nonce[crypto_secretbox_NONCEBYTES];
+
+  SodiumState() {
+    sodium_init(); // this can fail with a non-zero return code
+    crypto_secretbox_keygen(key);
+    randombytes_buf(nonce, sizeof nonce);
```

Comment:
See the other comments I have made about non-determinism.
I think it would be better to use `
randombytes_buf_deterministic` instead of 
`randombytes_buf`.